### PR TITLE
Packaging for Linux

### LINKED ISSUE #452: Linux automated builds for releasepage
Another one - it would be great to be able to prepare for Linux automated builds for the releasepage - eventually Linux will work pretty close to 100% so we're going to need this ticket i guess.

### PR DESCRIPTION
**Pull request by baconpaul**
Add scripts (authored originally by @rgvdberg) to make a debian
package from a built surge. Adjust paths accordingly.

Closes #452 but will require future better packaging, per conversations
with @falktx

Co-authored-by: Paul Walker <paul@pwjw.com>
Co-authored-by: Rob van den Berg <rghvdberg@gmail.com>

### CODE REVIEW

File: installer_linux/make_deb.sh [hunk 1]
```diff
@@ -0,0 +1,77 @@
+#!/bin/bash
+
+# preflight check
+
+if [[ ! -f ./make_deb.sh ]]; then
+	echo "You must run this script from within the installer_osx directory!"
```
Comment (kzantow) [inline installer_linux/make_deb.sh]: `installer_linux` :)

Comment (baconpaul) [reply installer_linux/make_deb.sh]: Oh thanks!

File: installer_linux/make_deb.sh [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+mkdir -p surge-synthesizer/DEBIAN
+
+# build control file
+
+if [[ -f surge-synthesizer/DEBIAN/control ]]; then
+	rm surge-synthesizer/DEBIAN/control
+fi
+touch surge-synthesizer/DEBIAN/control
+cat <<EOT >> surge-synthesizer/DEBIAN/control
+Package: surge-synthesizer
```
Comment (falkTX) [inline installer_linux/make_deb.sh]: package name should be just "surge", no?

**Comment by falkTX** [comment]
I do not see a changelog file being added to the debian file here.
this will make it being seen as a bad/unsafe debian package.

**Comment by baconpaul** [comment]
"surge" or "Surge" for the package name? I'm happy with either.

I'm not sure what to put in the changelog and how to generate it from a non-manual build. I found some folks who use git log but that's way too verbose. Ideas on what you would do?

File: src/common/SurgeStorage.cpp [hunk 3]
```diff
@@ -161,6 +161,16 @@ SurgeStorage::SurgeStorage()
    else
       datapath = std::string(homePath) + "/.local/share/Surge/";
```
Comment (kzantow) [inline src/common/SurgeStorage.cpp]: I think the linux way would be `/usr/share/surge` with lowercase, but this is fine for consistency with other platforms, too...

File: installer_linux/make_deb.sh [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    echo "DEB VERSION is ${DEB_VERSION}"
+fi
+
+
+# Names
+SURGE_NAME=Surge
+# SURGE_NAME=surge-synthesizer
+
+# locations
+
```
Comment (kzantow) [inline installer_linux/make_deb.sh]: maybe `rm -rf surge-synthesizer` to make sure a failed prior run doesn't leave bad stuff around?

File: installer_linux/make_deb.sh [hunk 5]
```diff
@@ -0,0 +1,77 @@
+#!/bin/bash
+
```
Comment (kzantow) [inline installer_linux/make_deb.sh]: `set -e`?

File: installer_linux/make_deb.sh (hunk 2 above)
Comment (baconpaul) [reply installer_linux/make_deb.sh]: Yup changed this in my force push now.

File: installer_linux/make_deb.sh (hunk 5 above)
Comment (baconpaul) [reply installer_linux/make_deb.sh]: Oh just saw this now. Yeah set -e would be good. Let me let this force push run first.

**Comment by baconpaul** [comment]
OK if this force push runs then I'll do another force push which takes the two @kzantow things I missed (a set -e; and a cleanup before I begin) and then force push and merge.

File: src/common/SurgeStorage.cpp (hunk 3 above)
Comment (baconpaul) [reply src/common/SurgeStorage.cpp]: Yeah lets leave it consistent for now and then revisit maybe later was my thinking.

File: installer_linux/make_deb.sh [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+cp ../LICENSE ${PACKAGE_NAME}/usr/share/${SURGE_NAME}/doc
+cp -r ../resources/data/* ${PACKAGE_NAME}/usr/share/${SURGE_NAME}/
+cp ../target/vst2/Release/Surge.so ${PACKAGE_NAME}/usr/lib/vst/${SURGE_NAME}.so
+
+# Once VST3 works, this will be ../products/vst3
+# cp ../target/vst3/Release/Surge.so ${PACKAGE_NAME}/usr/lib/vst3/${SURGE_NAME}.so
+
+#build package
+
+dpkg-deb --build ${PACKAGE_NAME}
```
Comment (kzantow) [inline installer_linux/make_deb.sh]: `dpkg-deb --build ${PACKAGE_NAME} product/${PACKAGE_NAME}-linux-x64-${VERSION}.deb` ?

Comment (baconpaul) [reply installer_linux/make_deb.sh]: Oh I guess yeah I can do that! Let me test and push

**Pull request merged**